Gem/Code/Source/Components/NetworkCoinComponent.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Source/Components/NetworkCoinComponent.h>
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <Multiplayer/Components/NetworkTransformComponent.h>
+
+namespace MultiplayerSample
+{
+    void NetworkCoinComponent::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serializeContext)
+        {
+            serializeContext->Class<NetworkCoinComponent, NetworkCoinComponentBase>()
+                ->Version(1);
+        }
+        NetworkCoinComponentBase::Reflect(context);
+    }
+
+    void NetworkCoinComponent::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+        if (IsNetEntityRoleClient())
```

Comment:
Could the displacement of the coin be significant enough that between min and max height a small character might be able to walk through it? You always need to be very careful with visually altering the transforms of anything on a client that can be interacted with on the server without making that same change on the server. That sort of desync can really catch you by surprise.

Reply:
Hmm, yeah, it could. The movement is configurable on the component and I am making it small enough where that won't happen. Otherwise, we would have to synchronize all the coin transforms. Yuck!

Reply:
synchronize the coin transforms, or just sync a quantized phase value that gets applied consistently on both endpoints.

Reply:
For what it's worth, we have QuantizedValues to make transmitting 1, 2, 3, or 4 floats across the network in 1, 2, 3, or 4 bytes per component nice and easy. So QuantizedValues<1, 2, -1, 1> would be a single float that can range from [-1, 1] and transmits using only two bytes.

Reply:
While we can drive coin spinning behavior from authority we can't mimic the same behavior on authority because then NetworkTransformComponent will push the position and rotation down to clients, which would defeat the purpose of building a custom replication here.

Now if NetworkTransformComponent had a way to disable replication of position and/or rotation and/or scale, then I could replicate coin rotation in more specialized manner.

Reply:
Cool example of having animation on the client without affecting gameplay (doesn't need network syncing).

Reply:
In this case you would have to make a visual-only tweak of the entities rendered position without actually touching the network transform representation itself.. I would have to deeply dig into code to see how easy this would be, if atom exposes a render proxy displacement setting or the mesh component on the coins themselves, or something similar..